pkg/manager/member/tikv_member_manager.go

```diff
@@ -223,7 +223,7 @@ func (tkmm *tikvMemberManager) syncStatefulSetForTidbCluster(tc *v1alpha1.TidbCl
 		return err
 	}
 
-	if tkmm.autoFailover {
+	if tkmm.autoFailover && tc.Spec.TiKV.MaxFailoverCount != nil {
```

Comment:
The pointer will never be nil after defaulting.

Reply:
I think the defaulting and checking are 2 different things.